### DISCUSSION TITLE
How to correctly create Enum object and use it in models in sqlalchemy 2.0?

### DISCUSSION THREAD

**Discussion by Vlad0n1m** [sqlalchemy/sqlalchemy#10615] [Usage Questions]
Hi there, I need to create Enum field for my model, but as I know sqlalchemy 2.0 uses Mapped, mapped_column etc. but in documentation of sqlalchemy for Enum objects they use the old version, so now im a bit conused how to do it properly... I would be very happy if someone gave me an advice how to create Enum class and how to pass it to the Model field

**Comment by zzzeek** [comment]
hi -

it might not be easy to find as it's kind of in a deep section but we have a whole explanation of using enums with the new declarative style here: https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#using-python-enum-or-pep-586-literal-types-in-the-type-map that should have everything you need

**Reply by CaselIT** [reply]
Should probably add a see-also in the enum documentation

**Reply by CaselIT** [reply]
nevermind, it's already there
